### PR TITLE
fix(TDI-43995): Support listing any path and file

### PR DESCRIPTION
**Pull request by OleksiiNimych**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-43995
The component tBoxList lists files from parent folder. When you pass the path to the file, component displays siblings folder on the same base path. When you pass path that does not exist displaying content of the root folder.

**What is the new behavior?**
Supported listing any path.
If the path is a file, only list the file.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [x] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tBoxList/tBoxList_begin.javajet [hunk 1]
```diff
@@ -105,115 +105,79 @@ if(!"true".equals(useExistingConn)){
...
-                    List<String> <%=cid%>_paths = new java.util.ArrayList<String>(java.util.Arrays.asList(parentPath.split("/")));
+	com.box.boxjavalibv2.dao.BoxFolder <%=cid%>_rootFolder = <%=cid%>_client.getFoldersManager().getFolder("0", null);
+	com.box.boxjavalibv2.dao.BoxItem <%=cid%>_boxItem;
 
-                    <%=cid%>_boxFolder = getBoxFolderRecursively(<%=cid%>_paths, <%=cid%>_boxFolder, <%=cid%>_client);
-                }
-			}
+	if (org.apache.commons.lang.StringUtils.substringBefore(<%=cid%>_path, "/").equals("All Files")) {
+		List<String> <%=cid%>_pathItems = new java.util.ArrayList<String>(java.util.Arrays.asList(<%=cid%>_path.split("/")));
+		<%=cid%>_pathItems.remove(0);
```
Comment (mbasiuk-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tBoxList/tBoxList_begin.javajet:120]: What if root folder was specified incorrectly?

Comment (mbasiuk-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tBoxList/tBoxList_begin.javajet:120]: /root/someFolder   vs /usr/someFolder ?
First element is removed from list, and someFolder == someFolder ?

File: main/plugins/org.talend.designer.components.localprovider/components/tBoxList/tBoxList_end.javajet [hunk 2]
```diff
@@ -16,7 +16,6 @@
 %>
     }
 		
-}
```
Comment (mbasiuk-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tBoxList/tBoxList_end.javajet:19]: it's hard to catch were it's missing. Be very careful.

File: main/plugins/org.talend.designer.components.localprovider/components/tBoxList/tBoxList_begin.javajet (hunk 1 above)
Comment (OleksiiNimych) [reply main/plugins/org.talend.designer.components.localprovider/components/tBoxList/tBoxList_begin.javajet:120]: As i understand the root folder at remote Box storage is "All Files". Root folder creates by default and I didn't found way to change it name. So root folder always "All Files". I check name of root folder at 118 row and if user specify another name for root folder an exception will be thrown.

File: main/plugins/org.talend.designer.components.localprovider/components/tBoxList/tBoxList_end.javajet (hunk 2 above)
Comment (OleksiiNimych) [reply main/plugins/org.talend.designer.components.localprovider/components/tBoxList/tBoxList_end.javajet:19]: This is a closing bracket of else if condition at https://github.com/Talend/tdi-studio-se/blob/master/main/plugins/org.talend.designer.components.localprovider/components/tBoxList/tBoxList_begin.javajet#L161-L162

**Pull request merged**